Compiler bug when improperly casting string to rune

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Linux version 5.16.0-arch1-1 (linux@archlinux) (gcc (GCC) 11.1.0, GNU ld (GNU Binutils) 2.36.1) #1 SMP PREEMPT Mon, 10 Jan 2022 20:11:47 +0000
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
CC version: cc (GCC) 11.1.0

getwd: /home/z/onedrive/Documents/Development/Projects/Testing/V
vmodules: /home/z/.vmodules
vroot: /home/z/v
vexe: /home/z/v/v
vexe mtime: 2022-01-17 10:19:27
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 93c40e6.727c9fb

Git version: git version 2.34.1
Git vroot status: weekly.2022.02-46-g727c9fb4
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg main.v`
```v
println(rune('0'))
```

**What did you expect to see?**

A compile error detailing a bad cast from string to rune


**What did you see instead?**
```
==================
/tmp/v_1000/main.542034426795663815.tmp.c:11057: error: cannot convert 'struct string' to 'unsigned int'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -2732,6 +2732,11 @@ pub fn (mut c Checker) cast_expr(mut node ast.CastExpr) ast.Type {
 			node.pos)
 	}
 
+	if to_sym.kind == .rune && from_sym.is_string() {
+		snexpr := node.expr.str()
+		c.error('cannot cast `$from_sym.name` to rune, use `${snexpr}.runes()` instead.', node.pos)
+	}
+
 	if to_type == ast.string_type {
 		if from_type in [ast.byte_type, ast.bool_type] {
 			snexpr := node.expr.str()

--- a/vlib/v/checker/tests/cast_string_to_rune_err.out
+++ b/vlib/v/checker/tests/cast_string_to_rune_err.out
@@ -1,24 +1,24 @@
-../err.v:2:7: error: cannot cast `string` to rune, use `'0'.runes()` instead.
+vlib/v/checker/tests/cast_string_to_rune_err.vv:2:7: error: cannot cast `string` to rune, use `'0'.runes()` instead.
     1 | fn main() {
     2 |     _ := rune('0')
       |          ~~~~~~~~~
     3 |     _ := rune('012345')
     4 |     _ := rune('v')
-../err.v:3:7: error: cannot cast `string` to rune, use `'012345'.runes()` instead.
+vlib/v/checker/tests/cast_string_to_rune_err.vv:3:7: error: cannot cast `string` to rune, use `'012345'.runes()` instead.
     1 | fn main() {
     2 |     _ := rune('0')
     3 |     _ := rune('012345')
       |          ~~~~~~~~~~~~~~
     4 |     _ := rune('v')
     5 |     _ := rune('abcd')
-../err.v:4:7: error: cannot cast `string` to rune, use `'v'.runes()` instead.
+vlib/v/checker/tests/cast_string_to_rune_err.vv:4:7: error: cannot cast `string` to rune, use `'v'.runes()` instead.
     2 |     _ := rune('0')
     3 |     _ := rune('012345')
     4 |     _ := rune('v')
       |          ~~~~~~~~~
     5 |     _ := rune('abcd')
     6 | }
-../err.v:5:7: error: cannot cast `string` to rune, use `'abcd'.runes()` instead.
+vlib/v/checker/tests/cast_string_to_rune_err.vv:5:7: error: cannot cast `string` to rune, use `'abcd'.runes()` instead.
     3 |     _ := rune('012345')
     4 |     _ := rune('v')
     5 |     _ := rune('abcd')

--- a/vlib/v/checker/tests/cast_string_to_rune_err.out
+++ b/vlib/v/checker/tests/cast_string_to_rune_err.out
@@ -1,0 +1,26 @@
+../err.v:2:7: error: cannot cast `string` to rune, use `'0'.runes()` instead.
+    1 | fn main() {
+    2 |     _ := rune('0')
+      |          ~~~~~~~~~
+    3 |     _ := rune('012345')
+    4 |     _ := rune('v')
+../err.v:3:7: error: cannot cast `string` to rune, use `'012345'.runes()` instead.
+    1 | fn main() {
+    2 |     _ := rune('0')
+    3 |     _ := rune('012345')
+      |          ~~~~~~~~~~~~~~
+    4 |     _ := rune('v')
+    5 |     _ := rune('abcd')
+../err.v:4:7: error: cannot cast `string` to rune, use `'v'.runes()` instead.
+    2 |     _ := rune('0')
+    3 |     _ := rune('012345')
+    4 |     _ := rune('v')
+      |          ~~~~~~~~~
+    5 |     _ := rune('abcd')
+    6 | }
+../err.v:5:7: error: cannot cast `string` to rune, use `'abcd'.runes()` instead.
+    3 |     _ := rune('012345')
+    4 |     _ := rune('v')
+    5 |     _ := rune('abcd')
+      |          ~~~~~~~~~~~~
+    6 | }

--- a/vlib/v/checker/tests/cast_string_to_rune_err.vv
+++ b/vlib/v/checker/tests/cast_string_to_rune_err.vv
@@ -1,0 +1,6 @@
+fn main() {
+	_ := rune('0')
+	_ := rune('012345')
+	_ := rune('v')
+	_ := rune('abcd')
+}

